### DISCUSSION TITLE
NextJS13 page to reload when switching between root layouts

### DISCUSSION THREAD

**Discussion by denisso** [vercel/community#1414] [Help]
Trying out new features Next13.

I found an the example of [Creating multiple root layouts](https://beta.nextjs.org/docs/routing/defining-routes#example-creating-multiple-root-layouts).
I wrote this [code](https://github.com/denisso/next13-app-exp/tree/111e991cc12876efd65f223b2b6aae8d9a78b366). As it is written in the documentation, I added <html> and <body> tags html and body to each layout.

Deployed my code on [vercel](https://multiple-root-layouts-e20pt24sg-denisso.vercel.app/).

I didn't find any mention in the documentation about whether the page should be reloaded during the transition between root layouts. I think it shouldn't, but in development mode, the page is reloaded every time the root layout route is changed, and in production mode, once out of three.

My question is: should the page be reloaded when switching between root layouts?

**Comment by amyegan** [comment]
Thanks for reaching out @denisso! Can you try updating your `next` version to `13.1.6` and let me know if that fixes it?

**Reply by denisso** [reply]
Thanks for the advice @amyegan. I've upgraded NextJS to 13.1.6(latest) but the page is still reloading

**Reply by amyegan** [reply]
Great news, Denis! You found an edge case, but the engineers have a solution. It's available now on canary, which you can install using `npm install next@canary`

**Reply by denisso** [reply]
I was glad to help, @amyegan!

**Comment by denisso** [comment]
Hello again @amyegan. I finally found the time to try the canary update. But in my opinion it only got worse, now the page is reloaded every time the Root Layout is changed. So it works on version [13.1.6](https://multiple-root-layouts-e20pt24sg-denisso.vercel.app/) so on version [13.1.7-canary](https://multiple-root-layouts-pv0yi955m-denisso.vercel.app/)

I found a new item [in the documentation for multiple layouts](https://beta.nextjs.org/docs/routing/defining-routes#good-to-know).

> Navigating across multiple root layouts will cause a full page load (as opposed to a client-side navigation)

Does this mean that a full page reload when changing the RootLayout is normal behavior and it will not work without reloading the page?
